### ISSUE TITLE
[Bug]: Generics trigger STALE_REFERENCE_ERROR

### ISSUE THREAD

**Issue by ckankare** [AdaCore/ada_language_server#1219]
### Environment

- OS and Version: WSL2 (under Windows 11), running Ubuntu 24.04
- VS Code Version: 1.94.2
- Ada & SPARK Extension Version: v25.0.20241014

### Bug Summary and Reproducer

Bug Summary:

Generics causes the language server to crash with
```
Message: Exception: raised LANGKIT_SUPPORT.ERRORS.STALE_REFERENCE_ERROR : related unit was reparsed
```
Steps to reproduce:

With the following file
```ada
procedure Main is

   generic

   type Value_T is private;

   package Foos is

      Value : Value_T;

   end Foos;

   package Bars is new Foos (Integer);

begin
    Bars.Value := 10;
end;
```
add a space somewhere (or something that triggers `textDocument/semanticTokens/full`), this causes the language server to crash with the following error:
```
[Error - 8:52:59 PM] Request textDocument/semanticTokens/full failed.
  Message: Exception: raised LANGKIT_SUPPORT.ERRORS.STALE_REFERENCE_ERROR : related unit was reparsed
```
and the following stack trace:
```
0x563b972166de Libadalang.Analysis.Check_Safety_Net.Localalias at libadalang-analysis.adb:36610
0x563b9725f146 Libadalang.Analysis.P_Decl at libadalang-analysis.adb:24221
0x563b9690bb9b Lsp.Ada_Highlighters.Highlight_Name at lsp-ada_highlighters.adb:571
0x563b9690d57c Lsp.Ada_Highlighters.Highlight_Node at lsp-ada_highlighters.adb:855
0x563b96acf734 Lsp.Ada_Tokens_Full.Execute_Ada_Request at lsp-ada_tokens_full.adb:114
0x563b95d49023 Lsp.Ada_Request_Jobs.Execute_Request at lsp-ada_request_jobs.adb:33
0x563b95d48df9 Lsp.Server_Request_Jobs.Execute at lsp-server_request_jobs.adb:46
0x563b95d228ff Lsp.Job_Schedulers.Process_Job at lsp-job_schedulers.adb:183
0x563b96a59d4a Lsp.Servers.Processing_Task_TypeT at lsp-servers.adb:847
0x563b98c1dbed system__tasking__stages__task_wrapper at ???
```

Attached reproducible case and ALS log file.
[als_bug_1.zip](https://github.com/user-attachments/files/17511048/als_bug_1.zip)
[ada_ls_log.2024-10-24T211228.log](https://github.com/user-attachments/files/17511177/ada_ls_log.2024-10-24T211228.log)

### Configuration and Logs

```shell
-
```

**Comment by ec-c** [comment]
I can confirm this bug using:
* OS: Darwin arm64 24.0.0
* VS Code Version: 1.94.2 (Universal)
* Ada & SPARK: v25.0.20241014

**Comment by reznikmm** [comment]
This is fixed in the development version (actually in libadalang)

**Comment by mgrojo** [comment]
When having this error, you get recurrent notifications:

![imagen](https://github.com/user-attachments/assets/d05f24cb-89b2-42e2-9892-6729644d41bb)

Apparently, you should be able to disable those notifications using the cogwheel, but they keep appearing after setting the notifications to Off. Is this a bug in this extension or in VS Code 1.95.2 for Linux?

**Comment by AnthonyLeonardoGracio** [comment]
> When having this error, you get recurrent notifications:
> 
> ![imagen](https://private-user-images.githubusercontent.com/13812910/386950214-d05f24cb-89b2-42e2-9892-6729644d41bb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5NDE0OTUsIm5iZiI6MTczMTk0MTE5NSwicGF0aCI6Ii8xMzgxMjkxMC8zODY5NTAyMTQtZDA1ZjI0Y2ItODliMi00MmUyLTk4OTItNjcyOTY0NGQ0MWJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE4VDE0NDYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ2YWEzOTUyNmE4NjA5NDMyZDA5YWUwMTc1N2I4YTExOWY0YzRlZjVmOWE4MDc5NDAyNzYwMDRjY2I2MjJjMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ya0YlFP93PLNTiPYoLL4VZ9GMh8mXB6nLru-f07WBIU)
> 
> Apparently, you should be able to disable those notifications using the cogwheel, but they keep appearing after setting the notifications to Off. Is this a bug in this extension or in VS Code 1.95.2 for Linux?

We have noticed that too, it seems like a bug in VS Code, but we'll create an internal issue to investigate why it does not work.

**Comment by reznikmm** [comment]
This should be fixed in [26.0.202411173](https://github.com/AdaCore/ada_language_server/releases/tag/26.0.202411173) release.
